app/models/postcode.rb

```diff
@@ -0,0 +1,12 @@
+class Postcode < ApplicationRecord
+  validates_with PostcodeValidator
+  validates :postcode, uniqueness: true
+  before_validation :normalize_postcode
+
+private
+
+  def normalize_postcode
+    self["postcode"] = self["postcode"].to_s.gsub(PostcodeValidator::DISALLOWED_CHARS, "").upcase
+    true
```

Comment:
Do you need to return `true` here?  I don't think the `before_validation` callback requires a `true` return value.

Reply:
I added the line based on [this recommendation](https://stackoverflow.com/a/2814037). But you're right - on second look, it doesn't seem to be needed:

```
"foo".gsub("bar", "")
=> "foo"

"foo".gsub!("bar", "")
=> nil
```

We use the `gsub` rather than the `gsub!` method, so will always return a string, which isn't falsey.